src/frameworks/next/index.ts

```diff
@@ -277,13 +277,19 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
   }
 
   const [middlewareManifest, prerenderManifest, routesManifest] = await Promise.all([
-    readJSON<MiddlewareManifest>(join(sourceDir, distDir, "server", MIDDLEWARE_MANIFEST)),
+    readJSON<MiddlewareManifest | MiddlewareManifestV1>(
```

Comment:
can we instead make `type MiddlewareManifest = MiddlewareManifestV1 | MiddlewareManifestV2`

Reply:
alt, if the middleware manifest type is exported by Next 12 we could add next 12 as a named devDep in the package.json and leverage that. e.g, `"next12": "npm:next@^12", "next13": "npm:next@^13"`

Reply:
I initially tried having `next12` as a separate dev dep but Next.js has some internal imports from `next/dist/...` that breaks the build. I can investigate further

Reply:
@leoortizz if you tried that and it didn't work well, let's punt. Not worth the time for Next 12, which we'll probably drop at some point